HALOEXCHANGE-OMPTarget references non-existent omp_target_parallel_exec policy

Trying to build the OpenMP Target version with the Cray Clang compiler, compilation fails during the HALOEXCHANGE app:

```
[  4%] Building CXX object src/CMakeFiles/raja-perf-omptarget.exe.dir/apps/HALOEXCHANGE-OMPTarget.cpp.o
/home/users/sabbott/nightly_crons/rajaperf-openmp-gfx906/RAJAPerf/src/apps/HALOEXCHANGE-OMPTarget.cpp:101:28: error: no type named 'omp_target_parallel_exec' in namespace 'RAJA'; did you mean 'omp_target_parallel_for_exec_nt'?
    using EXEC_POL = RAJA::omp_target_parallel_exec;
                     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
                           omp_target_parallel_for_exec_nt
/home/users/sabbott/nightly_crons/rajaperf-openmp-gfx906/RAJAPerf/tpl/RAJA/include/RAJA/policy/openmp_target/policy.hpp:79:20: note: 'omp_target_parallel_for_exec_nt' declared here
using policy::omp::omp_target_parallel_for_exec_nt;
                   ^
1 error generated.
```
This line seems to be at fault:

https://github.com/LLNL/RAJAPerf/blob/5f37acca141b52344a9916f6cb95c606287f0032/src/apps/HALOEXCHANGE-OMPTarget.cpp#L101

Changing that to the compiler-recommended `omp_target_parallel_for_exec_nt` does work.

For the record, I'm configuring with:
```
cmake \
         -DENABLE_ALL_WARNINGS=Off \
         -DENABLE_OPENMP=On \
         -DENABLE_TARGET_OPENMP=On \
         -DENABLE_CUDA=Off \
         -DENABLE_CLANG_CUDA=Off \
         -DENABLE_HIP=Off \
         -DENABLE_CLANG_HIP=Off \
         -DENABLE_TBB=Off \
         -DCMAKE_CXX_COMPILER=CC \
         -DCMAKE_C_COMPILER=cc \
         -DCMAKE_BUILD_TYPE=Release \
         -DCMAKE_CXX_STANDARD=c++11 \
         -DENABLE_TESTS=Off \
         -DENABLE_EXAMPLES=Off ../
```

--- a/src/apps/HALOEXCHANGE-OMPTarget.cpp
+++ b/src/apps/HALOEXCHANGE-OMPTarget.cpp
@@ -21,6 +21,11 @@ namespace rajaperf
 namespace apps
 {
 
+  //
+  // Define threads per team for target execution
+  //
+  const size_t threads_per_team = 256;
+
 #define HALOEXCHANGE_DATA_SETUP_OMP_TARGET \
   int hid = omp_get_initial_device(); \
   int did = omp_get_default_device(); \
@@ -98,7 +103,7 @@ void HALOEXCHANGE::runOpenMPTargetVariant(VariantID vid)
 
     HALOEXCHANGE_DATA_SETUP_OMP_TARGET;
 
-    using EXEC_POL = RAJA::omp_target_parallel_exec;
+    using EXEC_POL = RAJA::omp_target_parallel_for_exec<threads_per_team>;
 
     startTimer();
     for (RepIndex_type irep = 0; irep < run_reps; ++irep) {

